### ISSUE TITLE
Multi-outcome prediction investigation

### ISSUE THREAD

**Issue by alessandramaranca** [niznik-dev/cruijff_kit#82]
Think through and decide on an approach to evaluate and capture multi-outcome prediction with LLMs. Basically, our current evaluation pipeline does not work for more than one outcome (in particular, think about the way we are extracting the logits, or doing binary outcome prediction). I'm thinking we might need chat template and force the LLM to understand some kind of vector (for instance, the output always looks like _(x,y)_ where x and y are two predicands). Need to:

1. Conceptually sketch out fine-tuning and evaluation for multi outcome prediction. 
2. Implement a working minimal working pipeline. 

This might be complicated.

**Comment by msalganik** [comment]
@alessandramaranca This is a great question.  

Right now we predict one outcome at one time point.  I think there are three possible generalizations:
- multiple outcomes at one time point (e.g., y_it1, y_it2; 2 outcomes for person i at time t) 
- single outcome at multiple time points (e.g, y_{i,t,1}, y_{i,t+1,1}; 1 outcome at time t and t+1) [there are lots of time series models like this if the predictor data also includes time series of y_{i, *, 1}]
- multiple outcomes at multiple time points

I tried to think about what is the easiest data that allows us to explore all three.  Before reading more, please stop and think about it so we can have independent ideas.

The first thing I came up with height/weight data for children (ideally with some basic co-variates about the family).  This data has an interesting non-linear structure over time where there should be some covariance between the height and weight.  It also has clear baselines.  I found a few datasets that have this information for individual kids but they all require some degree of data access (and it might be complex to use them with commercial services).
- [Canada] https://onlinelibrary.wiley.com/doi/full/10.1002/oby.21158
- [Norway] https://www.nature.com/articles/s41598-021-84615-w (great data but complex access)

So for rapid prototyping what if we switch to some fully public data?  For that I was thinking of aggregate data such as:
- Statistics Netherlands API: https://opendata.cbs.nl/statline/portal.html?_la=en&_catalog=CBS. 
- Data commons https://docs.datacommons.org/what_is.html
I can't think of what is the natural version of height-weight over time but maybe you have some ideas from all your general exam readings?

I think the "scorers" part of Inspect AI should help us do the eval.

Happy to discuss at our next meeting.

**Comment by niznik-dev** [comment]
I guess this is the other dimension of #508 - are we still interested?

**Comment by niznik-dev** [comment]
<img src="https://raw.githubusercontent.com/niznik-dev/llm-personas/main/fusions/mattie/mattie-icon.png" width="36" align="absmiddle"> 🎻 **MxC** ✳️

Closing as part of the project archive. The question is still a good one, and Matt's three-way taxonomy in the thread above is the right starting point if anyone picks it up. The continuous-outcome half landed in #508; multi-outcome never got code in this repo. @alessandramaranca may have a local branch if this is revisited. Recorded in the handoff notes.

—MxC
